filament/backend/src/metal/MetalShaderCompiler.mm

```diff
@@ -111,9 +111,14 @@ bool isReady() const noexcept {
         id<MTLLibrary> library = nil;
         switch (program.getShaderLanguage()) {
             case ShaderLanguage::MSL: {
-                // By default, Metal uses the most recent language version.
                 MTLCompileOptions* options = [MTLCompileOptions new];
                 options.fastMathEnabled = YES;
+                // The docs for MTLCompileOptions say that Metal uses the most
+                // recent language version if it is not specified, but this is
+                // not always true: it is possible for the environment to be set
+                // up in a way where an older version is used by default. So we
+                // explicitly set the minimum version we require.
+                options.languageVersion = MTLLanguageVersion2_3;
```

Comment:
I think it's good to be explicit with the language version here, but this should match what we pass to spirv-cross while compiling materials:

https://github.com/google/filament/blob/ad27d48fd309e1e8940e1a23a95ec992f07c481c/libs/filamat/src/GLSLPostProcessor.cpp#L359-L369

I think it's safe to go with MSL 2.0 on mobile, and MSL 2.3 on desktop. You can use a `#if defined(FILAMENT_IOS)` check here.